### PR TITLE
bc: replace GNU bc with gavinhoward/bc

### LINKED ISSUE #7195: Package request: Replace gnu-bc with "bsd"-bc
The gnu-bc has not been updated since 2017, nor does it have current Posix compliance. This project https://github.com/gavinhoward/bc is updated and complies with recent Posix standard.

### PR DESCRIPTION
**Pull request by gavinhoward**
This would close #7195 if accepted.

I tried to use `build_style=configure` with `configure_script=./configure.sh` and `configure_args=-GM`, but it was not working, or installing in the correct place.

I ran `./xbps-src -Q pkg bc` to ensure that the test suite passed.

I ran `xlint` to remove any lint errors.

I put myself as the maintainer because I did not know if Juan RP wanted to maintain this.

This `bc` respects the environment variables for cross compilation used by Void, so it should build fine in any situation.

### CODE REVIEW

File: srcpkgs/bc/template [hunk 1]
```diff
@@ -1,46 +1,27 @@
...
+revision=1
 short_desc="An arbitrary precision numeric processing language"
-maintainer="Juan RP <xtraeme@voidlinux.org>"
-homepage="http://www.gnu.org/software/${pkgname}/"
-license="GPL-3"
-distfiles="${GNU_SITE}/${pkgname}/${pkgname}-${version}.tar.gz"
-checksum=62adfca89b0a1c0164c2cdca59ca210c1d44c3ffc46daf9931cf4942664cb02a
-disable_parallel_build=yes
+maintainer="Gavin D. Howard <gavin.d.howard@gmail.com>"
+license="BSD-2"
```
Comment (maxice8) [inline srcpkgs/bc/template]: Please use the SPDX short-identifier

**Comment by maxice8** [comment]
This is a nicer template i got, you can get some ideas from it.

https://github.com/void-linux/void-packages/pull/7202

**Comment by maxice8** [comment]
> Juan RP 

He is our lead dev that left 1+ year ago, it made the news in some places.

**Comment by gavinhoward** [comment]
@maxice8 your template uses bc 1.0. My template updates to 1.1.0, which was released on Saturday.

File: srcpkgs/bc/template [hunk 2]
```diff
@@ -1,46 +1,27 @@
...
 pkgname=bc
-version=1.07.1
-revision=3
-build_style=gnu-configure
-configure_args="--with-readline"
-hostmakedepends="ed flex"
-makedepends="readline-devel"
+version=1.1.0
+revision=1
 short_desc="An arbitrary precision numeric processing language"
```
Comment (maxice8) [inline srcpkgs/bc/template]: Please remove the article, just say Arbitrary

**Comment by maxice8** [comment]
I am aware of that

**Comment by chneukirchen** [comment]
Please realize this with alternatives= there is no need to drop GNU bc.

**Comment by gavinhoward** [comment]
@chneukirchen noted. @maxice8 is that what you would prefer to do?

**Comment by maxice8** [comment]
I generally dislike alternatives= for basic system stuff like sh and bc but in this case:

- Create a gh-bc (or call it whatever you want as long as it is not too big) package for this bc. Have it provides an alternatives to `bc`
- Modify `bc` to install gnu-bc and provide an alternatives to `bc`.

THe question now are:

- keep `bc` as `bc` or rename it to `gnu-bc`
- make `gh-bc` as default `bc`

**Comment by gavinhoward** [comment]
Okay, will do.

Just to make sure I read the manual right, should the alternatives should be something like the following?

```
alternatives=bc:/usr/bin/bc:/usr/bin/bc-gh
alternatives=dc:/usr/bin/dc:/usr/bin/dc-gh
```

(My `bc` switches between `bc` and `dc` based on the first letter of the executable name.)

**Comment by maxice8** [comment]
See the examples from busybox

**Comment by gavinhoward** [comment]
Closing this. I will reopen a new one that has a new package and sets both the current bc and my bc up as alternatives for `bc`.

**Pull request closed without merging**